core/src/main/java/bio/terra/pearl/core/service/address/SmartyInternationalAddressValidationService.java

```diff
@@ -0,0 +1,217 @@
+package bio.terra.pearl.core.service.address;
+
+import bio.terra.pearl.core.model.address.AddressComponent;
+import bio.terra.pearl.core.model.address.AddressValidationResultDto;
+import bio.terra.pearl.core.model.address.MailingAddress;
+import com.smartystreets.api.international_street.Candidate;
+import com.smartystreets.api.international_street.Components;
+import com.smartystreets.api.international_street.Lookup;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Validation client for international addresses. Due to the complexity of global
+ * address formats, does not support all countries, but only a verified subset of
+ * countries.
+ */
+@Component
+@Slf4j
+public class SmartyInternationalAddressValidationService implements AddressValidationService {
+
+    private SmartyClient client;
+
+    SmartyInternationalAddressValidationService(SmartyClient client) {
+        this.client = client;
+    }
+
+    // see: https://www.iban.com/country-codes
+    // if a country is given that is not one of these countries, then
+    // validation will always error with an IllegalArgumentException
+    public static final List<String> SUPPORTED_COUNTRIES = List.of(
+            "CA",
+            "GB",
+            "MX"
```

Comment:
That was my thought. Even if a country doesn't need code changes, it would still need to be reviewed that it works with the code and that we are handling its format correctly. I considered adding them to the database but slightly leaned towards hardcoding them; happy to change it either way though.

Reply:
yeah, let's just keep this for now -- it'll be faster to iterate

Reply:
@devonbush The discussion in demo about making the countries that show up at the top / configuring which ones are visible makes me think that maybe we should go ahead and engineer them to be a part of the database. What are your thoughts?